src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -160,6 +164,40 @@ def logical_meter(self) -> LogicalMeter:
             )
         return self._logical_meter
 
+    def consumer(self) -> Consumer:
+        """Return the consumer instance.
+
+        If a Consumer instance doesn't exist, a new one is created and returned.
+
+        Returns:
+            A Consumer instance.
+        """
+        from ..timeseries.consumer import Consumer
+
+        if self._consumer is None:
+            self._consumer = Consumer(
+                channel_registry=self._channel_registry,
+                resampler_subscription_sender=self._resampling_request_sender(),
+            )
+        return self._consumer
```

Comment:
Not sure if this is in scope of this PR, as it is using a pattern already present in the file, but we could use `@lru_cache` to do the caching for us.

Well, thinking about it, we don't really need a cache (as we want the object, once is created, to live forever), maybe we should create a `@lazy` decorator, or add one more dependency and use https://pypi.org/project/lazy/

Reply:
I'll try the suggested lazy library / decorator